client/src/modules/shopping-list/model/reducer.js

```diff
@@ -28,14 +28,21 @@ const shoppingLists = (state = {}, action) => {
         [action.payload.listId]: updatedList
       };
     }
-    case ShoppingListActionTypes.FETCH_PRODUCTS_SUCCESS: {
-      const updatedShoppingList = {
-        ...state[action.payload.listId],
-        products: action.payload.products
+    case ShoppingListActionTypes.ARCHIVE_SUCCESS: {
+      const archivedList = {
+        _id: action.payload.listId,
+        isArchived: action.payload.isArchived
       };
       return {
         ...state,
-        [action.payload.listId]: updatedShoppingList
+        [action.payload.listId]: archivedList
+      };
+    }
+    case ShoppingListActionTypes.RESTORE_SUCCESS:
```

Comment:
this case will be related with my previous comment about fetching back whole data instead of just toggling back isArchived property using listID. So maybe this case need to be replaced by separate case

Reply:
archiveList removed data from redux so we have to get it one more time